File: yacastl/yacas/JavaYacas/net/sf/yacas/ArrayClass.java
package net.sf.yacas;


class ArrayClass extends GenericClass
{
  public ArrayClass(int aSize,LispObject aInitialItem)
  {
    iArray = new LispPtrArray(aSize,aInitialItem);
  }
  public String Send(LispArgList aArgList)
  {
    return null;
  }
  public String TypeName()
  {
    return "\"Array\"";
  }

  public int Size()
  {
    return iArray.Size();
  }
  public LispObject GetElement(int aItem) throws Exception
  {
    LispError.LISPASSERT(aItem>0 && aItem<=iArray.Size());
    return iArray.GetElement(aItem-1).Get();
  }
  public void SetElement(int aItem,LispObject aObject) throws Exception
  {
    LispError.LISPASSERT(aItem>0 && aItem<=iArray.Size());
    iArray.SetElement(aItem-1,aObject);
  }
  LispPtrArray iArray;
}

File: tags/v1_3_4/src/arrayclass.cpp

#include "yacasprivate.h"
#include "arrayclass.h"

ArrayClass::ArrayClass(LispInt aSize,LispObject* aInitialItem) : iArray()
{
  iArray.ResizeTo(aSize);
  if (aInitialItem)
  {
    for (LispInt i=0;i<aSize;i++)
    {
      iArray[i] = (aInitialItem);
    }
  }
}


ArrayClass::~ArrayClass()
{
}

const LispChar * ArrayClass::Send(LispArgList& aArgList)
{
    return NULL;
}
const LispChar * ArrayClass::TypeName()
{
    return "\"Array\"";
}



